packages/nitro-protocol/test/contracts/ninja-nitro/SingleChannelAdjudicator/claim.test.ts

```diff
@@ -0,0 +1,337 @@
+          'SingleChannelAdjudicator.claim.gas.md',
+          name + ': conclude Guarantor',
+          gasUsed
+        );
+      } else guarantor.outcome = guarantorOutcome; // Set this so that the claim tx should revert in the way we expect
+      // Compute an appropriate allocation outcome for the target (using only ETH)
+      const allocation = [];
+      Object.keys(tOutcomeBefore).forEach(key =>
+        allocation.push({destination: key, amount: tOutcomeBefore[key]})
+      );
+      const targetOutcome: Outcome = [
+        {assetHolderAddress: constants.AddressZero, allocationItems: allocation},
+      ];
+
+      // DEPLOY TARGET CHANNEL
+      gasUsed = await target.deploy();
+      await writeGasConsumption(
+        'SingleChannelAdjudicator.claim.gas.md',
+        name + ': deploy Target',
+        gasUsed
+      );
+      // CONCLUDE TARGET CHANNEL
+      if (Object.keys(tOutcomeBefore).length > 0) {
+        gasUsed = await target.conclude(targetOutcome);
+        await writeGasConsumption(
+          'SingleChannelAdjudicator.claim.gas.md',
+          name + ': conclude Target',
+          gasUsed
+        );
+      }
+
+      const tx = target.claimTx(guarantor, indices);
+      // Call method in a slightly different way if expecting a revert
+      if (reason) {
+        await expectRevert(() => tx, reason);
+      } else {
+        const balancesBefore = await getBalances(payouts);
+        // Extract logs
+        const {events: eventsFromTx, gasUsed} = await (await tx).wait();
+        await writeGasConsumption(
+          'SingleChannelAdjudicator.claim.gas.md',
+          name + ': Target. claim ',
+          gasUsed
+        );
+
+        // Check new outcomeHash
+        const newAllocation = [];
+        Object.keys(tOutcomeAfter).forEach(key =>
+          newAllocation.push({destination: key, amount: tOutcomeAfter[key]})
+        );
+        const outcome: Outcome = [
+          {assetHolderAddress: constants.AddressZero, allocationItems: newAllocation},
+        ];
+        const expectedFingerprint = channelDataToStatus({
+          turnNumRecord: 0,
+          finalizesAt: target.finalizesAt,
+          outcome,
+        });
+        // Check fingerprint against the expected value
+        // NOTE that allocations for zero amounts are left in place
+        expect(await target.statusOf()).toEqual(expectedFingerprint);
+
+        const balancesAfter = await getBalances(payouts);
+        Object.keys(payouts).forEach(key =>
+          expect(BigNumber.from(balancesAfter[key])).toEqual(
+            BigNumber.from(balancesBefore[key]).add(BigNumber.from(payouts[key]))
+          )
+        );
+      }
+    }
+  );
+});
+
+async function getBalances(payouts: AssetOutcomeShortHand) {
+  const balances: Record<string, BigNumber> = {};
+  await Promise.all(
+    Object.keys(payouts).map(async key => {
+      balances[key] = await provider.getBalance(convertBytes32ToAddress(key));
+    })
```

Comment:
```suggestion
    Object.keys(payouts).map(async destination => {
      balances[destination] = await provider.getBalance(convertBytes32ToAddress(destination));
    })
```

As above. `destination` here is from the type definition of `AssetOutcomeShortHand`.

Reply:
eda43d0